src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs

```diff
@@ -71,7 +71,7 @@ internal partial class GrpcWorkerChannel : IRpcWorkerChannel, IDisposable
         private ConcurrentDictionary<string, ExecutingInvocation> _executingInvocations = new();
         private IDictionary<string, BufferBlock<ScriptInvocationContext>> _functionInputBuffers = new ConcurrentDictionary<string, BufferBlock<ScriptInvocationContext>>();
         private ConcurrentDictionary<string, TaskCompletionSource<bool>> _workerStatusRequests = new ConcurrentDictionary<string, TaskCompletionSource<bool>>();
-        private List<IDisposable> _inputLinks = new List<IDisposable>();
+        private ConcurrentBag<IDisposable> _inputLinks = new ConcurrentBag<IDisposable>();
```

Comment:
We do not rely on the order of items in the collection. (_List is ordered, but ConcurrentBag has no order_)

Reply:
I bet this is related to some handling I added a while back in https://github.com/Azure/azure-functions-host/pull/9339 (I wanted more telemetry). I have not investigated, but we'd occasionally see nullrefs during disposal when looping through `_inputLinks`. I bet there has always been a race there.